tests/test_raster.py

```diff
@@ -4008,3 +4059,67 @@ def test_ufunc_methods(self, method_str):
         #
         # assert np.ma.allequal(outputs_ma[0], outputs_rst[0].data) and np.ma.allequal(
         #             outputs_ma[1], outputs_rst[1].data)
+
+    @pytest.mark.parametrize(
+        "np_func_name", ufuncs_str_2nin_1nout + ufuncs_str_2nin_2nout + handled_functions_2in
+    )  # type: ignore
+    def test_raise_errors_2nin(self, np_func_name: str) -> None:
+        """Check that proper errors are raised when input raster/array don't match (only 2-input functions)."""
+
+        # Create Rasters
+        ma = np.ma.masked_array(data=self.arr1, mask=self.mask1)
+        ma_wrong_shape = np.ma.masked_array(data=self.arr_wrong_shape, mask=self.mask_wrong_shape)
+        rst = gu.Raster.from_array(ma, transform=self.transform, crs=4326, nodata=_default_nodata(ma.dtype))
+        rst_wrong_shape = gu.Raster.from_array(
+            ma_wrong_shape, transform=self.transform, crs=4326, nodata=_default_nodata(ma_wrong_shape.dtype)
+        )
+        rst_wrong_crs = gu.Raster.from_array(ma, transform=self.transform, crs=32610, nodata=_default_nodata(ma.dtype))
+        rst_wrong_transform = gu.Raster.from_array(
+            ma, transform=self.wrong_transform, crs=4326, nodata=_default_nodata(ma_wrong_shape.dtype)
+        )
+
+        # Get ufunc
+        np_func = getattr(np, np_func_name)
+
+        # Strange errors happening only for these 4 functions...
```

Comment:
Maybe just refer to the opened issue here?

Reply:
Done!